components/camel-file/src/main/java/org/apache/camel/component/file/FileOperations.java

```diff
@@ -121,7 +127,14 @@ protected boolean buildDirectory(File dir, Set<PosixFilePermission> permissions,
             File base;
             // reusing absolute flag to handle relative and absolute paths
             if (absolute) {
-                base = new File("");
+                if (isWindows() && dir.getPath().startsWith(File.separator + File.separator)) {
+                    List<String> uncParts = Arrays.stream(parts).filter(s -> !s.isEmpty()).collect(Collectors.toList());
```

Comment:
this code is a bit hacky. 
Also can you show an UNC path example that currently fails.

Also mind there is camel-smb for SMB file systems.

Reply:
commit is reversed